orchestra/actions/install.py

```diff
@@ -125,14 +106,59 @@ def _prepare_tmproot(self):
             """)
         run_script(script, environment=self.environment)
 
-    def _cleanup_tmproot(self):
-        run_script('rm -rf "$TMP_ROOT"', environment=self.environment)
+    def _install_from_binary_archive(self, args):
+        # TODO: handle nonexisting binary archives
+        logger.info("Fetching binary archive")
+        self._fetch_binary_archive()
+        logger.info("Extracting binary archive")
+        self._extract_binary_archive()
 
-    def _install(self, quiet, test):
+        logger.info("Removing conflicting files")
+        self._remove_conflicting_files()
+
+    def _fetch_binary_archive(self):
+        # TODO: better edge-case handling, when the binary archive exists but is not committed into the
+        #   binary archives git-lfs repo (e.g. it has been locally created by the user)
+        binary_archive_path = self._binary_archive_filepath()
+        binary_archive_repo_dir = os.path.dirname(binary_archive_path)
+        while binary_archive_repo_dir != "/":
+            if ".git" in os.listdir(binary_archive_repo_dir):
+                break
+            binary_archive_repo_dir = os.path.dirname(binary_archive_repo_dir)
+        if binary_archive_repo_dir == "/":
+            logger.error("Binary archives are not a git repository!")
+            exit(1)
+        git_lfs.fetch(binary_archive_repo_dir, only=[os.path.realpath(binary_archive_path)])
+
+    def _extract_binary_archive(self):
+        if not self._binary_archive_exists():
+            raise Exception("Binary archive not found!")
+
+        archive_filepath = self._binary_archive_filepath()
+        script = dedent(f"""
+            mkdir -p "$TMP_ROOT$ORCHESTRA_ROOT"
+            cd "$TMP_ROOT$ORCHESTRA_ROOT"
+            tar xaf "{archive_filepath}"
+            """)
+        run_script(script, environment=self.environment, quiet=True)
+
+    def _implicit_dependencies(self):
+        if self.allow_binary_archive and self._binary_archive_exists():
+            return set()
+        else:
+            return {self.build.configure}
+
+    def _build_and_install(self, args):
         logger.info("Executing install script")
-        env = dict(self.environment)
-        env["RUN_TESTS"] = "1" if (self.build.test and test) else "0"
```

Comment:
This change broke a very important feature.
Please try to be much more careful when rebasing.

Reply:
Fixed in e5354de